### PR TITLE
Space type updates

### PR DESCRIPTION
**Pull request by jrmaddison**
Updates fenics_ice for https://github.com/jrmaddison/tlm_adjoint/pull/123

### CODE REVIEW

File: fenics_ice/solver.py [hunk 1]
```diff
@@ -858,7 +858,7 @@ def H_M_0(*X):
 
             M_inv_action = []
             for i, x in enumerate(X):
-                this_action = function_new(x, name=f"M_inv_action_{i}")
+                this_action = function_new_conjugate_dual(x, name=f"M_inv_action_{i}")
```
Comment (bearecinos) [inline fenics_ice/solver.py:861]: would this cause conflicts with what I have in the smith_glacier repo:

https://github.com/bearecinos/smith_glacier/blob/ce361ed4e5efde719d95bca08f9a20488a51e33e/meshtools/meshtools.py#L1055-L1056

I guess I would have to change this too to do the same. I do use the same function to compute vertex values for functions that require matrix multiplication or not:
```
  from tlm_adjoint.interface import function_new

    this_action = function_new(dV, name=f"M_inv_action")

    M_solver.solve(this_action.vector(), dV.vector())

    if mult_mmatrix:
        dV_alpha, dV_beta = this_action.split(deepcopy=True)
    else:
        dV_alpha, dV_beta = dV.split(deepcopy=True)

    # Vector to plot
    va = dV_alpha.compute_vertex_values(mesh)
    vb = dV_beta.compute_vertex_values(mesh)

    return va, vb
```

Comment (jrmaddison) [reply fenics_ice/solver.py:861]: Being precise you should use `function_new_conjugate_dual` for the case `mult_mmatrix = False` -- although in this case we don't have vertex values in the usual sense.

Comment (jrmaddison) [reply fenics_ice/solver.py:861]: Note also that space types are used only for consistency checking -- and will only raise warnings if inconsistencies are encountered.

Comment (jrmaddison) [reply fenics_ice/solver.py:861]: > Being precise you should use `function_new_conjugate_dual` for the case `mult_mmatrix = False` -- although in this case we don't have vertex values in the usual sense.

Sorry, this is incorrect, `this_action` is still a (primal space) function -- it is `dV` which is a (conjugate) dual space object. The code is fine as-is.

Comment (bearecinos) [reply fenics_ice/solver.py:861]: cool Thanks @jrmaddison ! Lets merge this then, just to double check I can keep using function_new() right?

Comment (jrmaddison) [reply fenics_ice/solver.py:861]: In most cases that will be correct.

**Review by bearecinos — APPROVED** [review]

**Pull request merged**